test/dummy.go

```
@@ -2,45 +2,66 @@
 	return e.stateRoot, e.maxBytes, nil
 }
 
 // GetTxs returns the list of transactions (types.Tx) within the DummyExecutor instance and an error if any.
 func (e *DummyExecutor) GetTxs(context.Context) ([]types.Tx, error) {
-	return e.txs, nil
+	txs := e.injectedTxs
+	e.injectedTxs = nil
+	return txs, nil
+}
+
+func (e *DummyExecutor) InjectTx(tx types.Tx) {
```

golangci-lint

[warning] exported: exported method DummyExecutor.InjectTx should have comment or be unexported

(revive)